### PR TITLE
Provide a custom task date formatter

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

- task due today or in the past shows "due today"
- task due in the next few days shows "x days left"
- task due long in the future shows nothing

medic/cht-core#5776

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by garethbowen** [comment]
Review please @dianabarsan . I thought this would be simpler than this, but I think it's working... Thanks for your time.

File: webapp/src/templates/directives/content_row_list_item.html [hunk 1]
```diff
@@ -28,9 +28,9 @@ <h4>
           <i class="fa fa-star primary {{primaryContact ? '' : 'ng-hide'}}" translate translate-attr-title="Primary contact"></i>
           <span>{{heading}}</span>
         </h4>
-        <div class="date {{due ? 'due' : ''}}">
```
Comment (dianabarsan) [inline webapp/src/templates/directives/content_row_list_item.html:31]: I believe we have some css about this that can be safely removed now that we're not using this class anymore: 
https://github.com/medic/cht-core/blob/master/webapp/src/css/content-list.less#L34

File: webapp/tests/karma/unit/services/relative-date.js [hunk 2]
```diff
@@ -223,6 +210,6 @@ describe('RelativeDate Service', () => {
     chai.expect(formatDateRelativeTime.callCount).to.equal(3);
     chai.expect(formatDateTime.callCount).to.equal(1);
     chai.expect(formatDateRelativeDay.callCount).to.equal(1);
-    done();
   });
+
```
Comment (dianabarsan) [inline webapp/tests/karma/unit/services/relative-date.js:214]: Consider adding unit tests for the new `getRelativeDate` exported function.

File: webapp/src/templates/directives/content_row_list_item.html (hunk 1 above)
Comment (garethbowen) [reply webapp/src/templates/directives/content_row_list_item.html:31]: Nice catch! Removed.

File: webapp/tests/karma/unit/services/relative-date.js (hunk 2 above)
Comment (garethbowen) [reply webapp/tests/karma/unit/services/relative-date.js:214]: Thank you! Done.

**Pull request merged**